src/NuGet.Clients/NuGet.VisualStudio/SolutionRestoreManager/IVsTargetFrameworkInfo.cs

```diff
@@ -21,17 +23,17 @@ public interface IVsTargetFrameworkInfo
         /// <summary>
         /// Collection of project references.
         /// </summary>
-        IVsReferenceItems ProjectReferences { get; }
+        IVsReferenceItems? ProjectReferences { get; }
```

Comment:
This gets into a more philosophical decision. Much like deserializing input, this gets provided to us by an external component. While we trust dotnet/project-system to do the right thing, we can't prevent 3rd party VS project systems from trying to call this API as well. If they don't enable nullable in their own repo and don't set this value to a non-null value, if we don't handle the nulls in our own code, then we'll throw a NullReferenceException and it might be challenging to determine the root cause.  If we say this could be null and we handle the null in our own code, then we won't get unhandled exceptions.

Additionally, there's a question about performance. If we force this collection to be non-null, even when there are no project references, package references, etc, then it forces the project system to make allocations. If we allow these properties to be null when there are zero items to report, then project system has the choice to not allocate. If they do allocate, and the VS Perf team complain about zero length collections, at least NuGet isn't causing the allocations through our design.